lib/ansible/vars/hostvars.py

```diff
@@ -75,7 +74,7 @@ def raw_get(self, host_name):
         '''
         host = self._find_host(host_name)
         if host is None:
-            return Undefined(name="hostvars['%s']" % host_name)
+            return AnsibleUndefinedVariable(name="hostvars['%s']" % host_name)
```

Comment:
`AnsibleUndefinedVariable` is an exception class. You probably want `AnsibleUndefined` which is Ansible's subclass of Jinja2's `Undefined`. See https://github.com/ansible/ansible/blob/bdd0fac606265f92e6521540106accf19a1ae4d0/lib/ansible/template/__init__.py#L232-L246

Reply:
yes your right, also the error msg is better. Changed it.